plugins/kotlin/j2k/shared/src/org/jetbrains/kotlin/nj2k/conversions/JavaStatementConversion.kt

```diff
@@ -167,4 +169,69 @@ class JavaStatementConversion(context: NewJ2kConverterContext) : RecursiveConver
                 tryStatement.catchSections.flatMap(::convertCatchSection)
             )
         ).withFormattingFrom(tryStatement)
+
+    private fun convertGuardStatement(ifElseStatement: JKIfElseStatement): JKStatement {
+        val thenExpression = ifElseStatement.thenBranch.statements.singleOrNull() ?: return ifElseStatement
+        if (thenExpression !is JKExpressionStatement) return ifElseStatement
+
+        val thrownExpression = thenExpression.expression.safeAs<JKThrowExpression>()?.exception ?: return ifElseStatement
+        if (thrownExpression !is JKNewExpression ||
+            thrownExpression.arguments.arguments.size > 1
+        ) {
+            return ifElseStatement
+        }
+
+        val expressionComparedToNull = ifElseStatement.condition.expressionComparedToNull()
+        val exceptionName = thrownExpression.identifier?.name
+        val correspondingMethodName = when (exceptionName) {
+            "IllegalArgumentException" -> if (expressionComparedToNull != null) "kotlin.requireNotNull" else "kotlin.require"
+            "IllegalStateException" -> if (expressionComparedToNull != null) "kotlin.checkNotNull" else "kotlin.check"
+            else -> return ifElseStatement
+        }
+        val exceptionArgument = thrownExpression.arguments.arguments.firstOrNull()
+        if (exceptionArgument != null && exceptionArgument.value !is JKLiteralExpression) {
```

Comment:
The original inspection instead checks that argument is of type String. Should we preserve this logic?

1. It may work not only with literal strings
2. It will future-proof the code against potential new overloads of the Java exception constructors.